File: ISAProject/src/main/java/com/app/restaurant/RestaurantService.java
package com.app.restaurant;

import com.app.drinkgrocery.DrinkGrocery;
import com.app.foodStuff.FoodStuff;

public interface RestaurantService {

	Iterable<Restaurant> findAll();
	Restaurant findOne(Long id);
	Restaurant save(Restaurant restaurant);
	void delete(Long id);
	Iterable<FoodStuff> getFoodStuff(Restaurant id);
	Iterable<DrinkGrocery> getDrinkGroceries(Restaurant id);
	Iterable<Restaurant> findRestaurantsWithoutManagers();
}

File: ISAProject/src/main/webapp/restaurant/restaurantService.js
app.service('restaurantService', ['$http', function($http){


    var url='/restaurants';
	
	this.findAll = function(){
		return $http.get(url);
	}
    
    this.findOne = function(id){
		return $http.get(url+"/"+id);
	}
    
    this.save = function(resId){
		return $http.post(url, resId);
	}

    this.getFoodStuff = function(id){
		return $http.get(url+"/getFoodStuff/"+id);
	}
    this.getDrinkGroceries = function(id){
		return $http.get(url+"/getDrinkGroceries/"+id);
	}
    this.findRestaurantsWithoutManagers = function(){
		return $http.get(url+"/findRestaurantsWithoutManagers");
	}
}]);